core/chapters/c06_lists.py

```diff
@@ -419,9 +419,140 @@ class index_error(Step):
         def check(self):
             return "IndexError" in self.result
 
-    class introducing_len_and_range(VerbatimStep):
+    class improve_with_list_and_loop(VerbatimStep):
         """
 There you go. `words[4]` and beyond don't exist, so trying that will give you an error.
+That first program is a bit repetitive. Let's improve it with a list and a loop!
+
+__program_indented__
+        """
+
+        def program(self):
+            words = ['This', 'is', 'a', 'list']
+            indices = [0, 1, 2, 3]
+
+            for index in indices:
+                print(index)
+                print(words[index])
+
+    class using_range_instead_of_indices(VerbatimStep):
+        """
+That's a bit better, but writing out `[0, 1, 2, ...]` isn't great, especially if it gets long.
+There's a handy function `range` to do that part for you. Replace `[0, 1, 2, 3]` with `range(4)`.
```

Comment:
```suggestion
There's a handy function `range` to do that part for you. Replace `[0, 1, 2, 3]` with `range(4)`,
i.e. `indices = range(4)`.
```

Gotta be extra careful with these instructions so people don't get stuck.